### ISSUE TITLE
meterSigGrp

### ISSUE THREAD

**Issue by craigsapp** [rism-digital/verovio#1268]
Somewhat related to issue https://github.com/rism-ch/verovio/issues/554, it would be useful to have "double" time signatures that indicate an alternating meter every other measure.

An exemplar would be from Benjamin Britten's _War Requium_ (mvmt. 6):

<img width="1802" alt="Screen Shot 2020-01-18 at 20 29 15" src="https://user-images.githubusercontent.com/3487289/72674963-64ac2300-3a32-11ea-8003-691ea3001769.png">

The duration of the measures alternates between 2/2 and 3/2 meters.

A related example would be Debussy preludes, book I, prelude X:

<img width="1221" alt="Screen Shot 2020-01-18 at 20 39 44" src="https://user-images.githubusercontent.com/3487289/72674987-bb196180-3a32-11ea-8426-1973efa764cc.png">

In this example the meaning is almost the same as the first example, except that it is hard to distinguish between the two meters since they have the same written durations.  But note the equals sign which is even more similar to issue #554 

-----------------------

For the first example, I am wondering if repeating the `<meterSig>` element with the two time signatures is a good method of encoding (although it is no purely semantic):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2020-01-18T20:33:06" version="2.4.0-dev-56dda9d-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv-0000001828529541">
    <score xml:id="score-0000000741013182">
     <scoreDef xml:id="scoredef-0000001434788564" midi.bpm="400">
      <staffGrp xml:id="staffgrp-0000002083486214">
       <staffDef xml:id="staffdef-0000000317997379" n="1" lines="5">
        <clef xml:id="clef-0000000057078570" shape="G" line="2" />
        <keySig xml:id="keysig-L3F1" sig="3s" />
        <meterSig xml:id="metersig-L2F1" count="2" unit="2" />
        <meterSig xml:id="metersig-L2F1b" count="3" unit="2" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1" n="0">
       <staff xml:id="staff-0000000720293967" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <note xml:id="note-L5F1" dur="1" oct="4" pname="a" accid.ges="n" />
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

The second meterSig is currently ignored in verovio rendering:

<img width="606" alt="Screen Shot 2020-01-18 at 20 43 54" src="https://user-images.githubusercontent.com/3487289/72675016-42ff6b80-3a33-11ea-9b1e-aeb8b971f910.png">

**Comment by rettinghaus** [comment]
That’s what meterSigGrp is for.

https://music-encoding.org/guidelines/v4/elements/metersiggrp.html

--
Dr. Klaus Rettinghaus

**Comment by craigsapp** [comment]
Yes, that is what is needed.   In this case it is an "alternating" pair of time signatures (for the first example).

Is there a way indicating the "=" sign for the Debussy example (which is an "interchanging" example).

In both cases for analysis (and performance of the Debussy), it would be useful to have an "invisible" time signature in each measure that describes the meter for that measure.

Current rendering:

<img width="442" alt="Screen Shot 2020-01-19 at 07 26 27" src="https://user-images.githubusercontent.com/3487289/72683542-24c65980-3a8d-11ea-89a5-8a8400554234.png">

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2020-01-18T20:33:06" version="2.4.0-dev-56dda9d-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv-0000001828529541">
    <score xml:id="score-0000000741013182">
     <scoreDef xml:id="scoredef-0000001434788564" midi.bpm="400">
      <staffGrp xml:id="staffgrp-0000002083486214">
       <staffDef xml:id="staffdef-0000000317997379" n="1" lines="5">
        <clef xml:id="clef-0000000057078570" shape="G" line="2" />
        <keySig xml:id="keysig-L3F1" sig="3s" />
        <meterSigGrp>
            <meterSig xml:id="metersig-L2F1" count="2" unit="2" />
            <meterSig xml:id="metersig-L2F1b" count="3" unit="2" />
        </meterSigGrp>
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1" n="0">
       <staff xml:id="staff-0000000720293967" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <note xml:id="note-L5F1" dur="1" oct="4" pname="a" accid.ges="n" />
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

**Comment by craigsapp** [comment]
Related to this is: how to deal with timestamp units when there is a meterSigGroup?  In other words, suppose that the meterSigGroup has two time signatures with different `@meter.unit` values, such as the _cathédrale engloutie_ example:


<img width="1221" alt="Screen Shot 2020-01-18 at 20 39 44" src="https://user-images.githubusercontent.com/3487289/72674987-bb196180-3a32-11ea-8426-1973efa764cc.png">


In this case the 6/4 has `@meter.unit="4"` and 3/2 has `@meter.unit="2"`.  Which unit would be used to describe timestamps in the music in such cases?   In this case the measure duration of the two time signatures is the same, so things can be faked by pretending that there is only one time signature.

However, if it were important to preserve the time signatures within the timestamps, this seems to be a problem, or at least an unnecessary complication because the timestamps have context-dependent units.

Here is a demonstration of the changeability of the timestamps:

<img width="538" alt="Screen Shot 2020-01-24 at 08 41 15" src="https://user-images.githubusercontent.com/3487289/73086347-596b5400-3e85-11ea-82f1-d3c119b45a4e.png">

The `@tstamp` units are "4":

```xml
<measure n="1">
   <staff n="1">
      <layer n="1">
         <note dur="4" oct="4" pname="c" />
         <note dur="4" oct="4" pname="d" />
         <note dur="4" oct="4" pname="e" />
         <note dur="4" oct="4" pname="f" />
      </layer>                  
   </staff>                 
   <fermata staff="1" tstamp="1" />
   <fermata staff="1" tstamp="2" />
   <fermata staff="1" tstamp="3" />
   <fermata staff="1" tstamp="4" />
</measure>    
```

<img width="531" alt="Screen Shot 2020-01-24 at 08 43 21" src="https://user-images.githubusercontent.com/3487289/73086510-9e8f8600-3e85-11ea-9cc6-9ea329b16713.png">

The timestamp units are now "8", so the values for the fermatas must be doubled:

```xml
<measure n="1">
   <staff n="1">
      <layer n="1">
         <note dur="4" oct="4" pname="c" />
         <note dur="4" oct="4" pname="d" />
         <note dur="4" oct="4" pname="e" />
         <note dur="4" oct="4" pname="f" />
      </layer>                  
   </staff>                 
   <fermata staff="1" tstamp="1" />
   <fermata staff="1" tstamp="3" />
   <fermata staff="1" tstamp="5" />
   <fermata staff="1" tstamp="7" />
</measure>     
```

A problem will occur if you change the time signature, but do not update the timestamps in the music:

Switching from 8/8 to 4/4 without adjusting the fermata timestamps:

<img width="523" alt="Screen Shot 2020-01-24 at 08 42 57" src="https://user-images.githubusercontent.com/3487289/73086627-d1397e80-3e85-11ea-98bf-dc51e9fc82a5.png">

Switching from 4/4 to 8/8 without adjusting the fermata timestamps:

<img width="521" alt="Screen Shot 2020-01-24 at 08 45 27" src="https://user-images.githubusercontent.com/3487289/73086684-ee6e4d00-3e85-11ea-8bfc-cdbb0750ee0b.png">

(I encode with `@startid` when possible to avoid the potential problem).

There is a discussion about this for future enhancement of MEI in such cases: https://github.com/music-encoding/music-encoding/issues/322

**Comment by rettinghaus** [comment]
implemented with #2271

**Comment by lpugin** [comment]
Yes, I think we can close it. The MEI clarification was essentially about the timestamps, but we can leave this for later to be discussed in the MEI repo
